openslides_backend/action/actions/assignment_candidate/mixins.py

```diff
@@ -29,8 +29,7 @@ def check_permissions(self, instance: Dict[str, Any]) -> None:
         if assignment.get("phase") == "voting":
             permission = Permissions.Assignment.CAN_MANAGE
             if not has_perm(self.datastore, self.user_id, permission, meeting_id):
-                msg = f"You are not allowed to perform action {self.name}."
-                msg += f" Missing permission: {permission}"
+                msg = f"Missing permission: {permission}"
                 raise PermissionDenied(msg)
```

Comment:
You can generalize it even more by adding a constructor to `PermissionDenied` which takes a permission (or Management Level) and the calculate the errro message based on that directly. If a string is given, it is just set directly. Note that `Permission` inherits from `str`, so you have to check for `Permission` first in the constructor.

Reply:
Then, most of the error raises become one-liners.

Reply:
There are also "Forbidden fields" permission denies or OrganisationLevel permission denies.
This would result in a complex __init__ to save one line. Too complex. Wouldn't do.

Reply:
With inheritance, you also solve this. `MissingPermission` gets the constructor described above and `PermissionDenied` can be used in all other cases.

Reply:
Use a `MissingPermission` exception to calculate the message in the missing case.
It inherits from PermissionDenied and needs a permission and the action name.